debugging-errors.mdx

```diff
@@ -0,0 +1,120 @@
+---
+
+We've compiled some of the common errors that people come across, as well as some approaches to debug these errors
+and the most common root causes. If you are still struggling to debug, please reach out in your support channel!
+
+## Query Errors
+
+### Unexpected query results
+
+If you're receiving either 0 rows when you expect a non-zero query result, or if the feature values that you're seeing
+in your query results do not match what you're expecting, we recommend that you follow the steps in
+[our guide for debugging queries](https://docs.chalk.ai/docs/debugging-queries). This includes some common tips and
+tricks, such as using `store_plan_stages` and stepping through the query plan.
+
+### Unexpected Chalk server error with status code 401
+
+The 401 status code means `unauthenticated`. We'd recommend verifying that the token you're using to run the query is
+correct. To refresh your token, you can also run `chalk login` from the terminal using our CLI and then `chalk config`
+to verify that your token has been provisioned for the right environment.
+
+### Unexpected Chalk server error with status code 500
+
+The 500 status code means `internal server error`. This should usually come with another error with more details on
+the root cause of the error. If you are struggling to debug this or find any other clues, please reach out to us!
+
+### Unexpected Chalk server error with status code 502
+
+The 502 status code means `bad gateway`, which usually indicates `timeout` or `not enough servers`. We'd recommend
+looking at CPU utilization on your cluster to determine if you need to modify the cloud resource configurations. You
+can do so in the `Settings > Resources` page in the Chalk console by modifying and applying resource configuration JSON
+changes under `Advanced Resource Configuration`. However, we'd also encourage reaching out to us for discussion on how
+to optimize resource configuration for your situation!
+
+### Query Timeout / Unclear Failure
+
+If you are running a query from a notebook or somewhere locally, the polling for query status might have a shorter
+timeout, so if the query times out before completion, we'd recommend verifying whether the query is still running in the
+console by looking at the query run page and validating that there is a pod in the cluster running the query. Otherwise
+your query may have also completed after the polling timeout, in which case you can also view query status in the
+console.
+
+### Please double check that all features have resolvers and resolvers do not have circular dependencies.
+
+The first thing to validate is that all features have resolvers. If all features have resolvers, then you might see
+this error message if Chalk is unable to construct a dependency graph for resolving the features that you are
+querying. This is usually a result of a join that cannot be processed. Please identify which feature(s) is triggering
```

Comment:
something like -- "we recommend reducing the number of output features in your query until you have a minimal set which still produces this error."